### PR TITLE
[Bot-Integration] Fix CloudAdapterTests

### PR DESCRIPTION
**Pull request by Batta32**
Fixes #

## Specific Changes

  - 
  -
  -

### CODE REVIEW

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/CloudAdapterBase.java [hunk 1]
```diff
@@ -108,8 +110,9 @@ protected CloudAdapterBase(BotFrameworkAuthentication withBotFrameworkAuthentica
                 logger.info(String.format("Sending activity. ReplyToId: %s", activity.getReplyToId()));
 
                 if (activity.isType(ActivityTypes.DELAY)) {
-                    int delayMs = Integer.valueOf((String) activity.getValue()) != null
-                        ? (int) activity.getValue() : DEFAULT_MS_DELAY;
+                    int delayMs = activity.getValue() != null
+                        ? new Double(String.valueOf(activity.getValue())).intValue()
```
Comment (ldardick) [inline libraries/bot-builder/src/main/java/com/microsoft/bot/builder/CloudAdapterBase.java]: Out of curiosity: Can't the value be directly parsed to `int` instead of parsing it as `double` and getting the integer-value?

**Review by ldardick — COMMENTED** [review]
All tests passed 

image

**Comment by luislera** [comment]
Tests passed successfully

imagen

**Review by luislera — APPROVED** [review]

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/CloudAdapterBase.java (hunk 1 above)
Comment (Batta32) [reply libraries/bot-builder/src/main/java/com/microsoft/bot/builder/CloudAdapterBase.java]: I'll be confirming this as it's really weird

Comment (Batta32) [reply libraries/bot-builder/src/main/java/com/microsoft/bot/builder/CloudAdapterBase.java]: I improved it parsing the Object value as Number (because Long/Float/Int/Double are Numbers) and then getting the `intValue` of the number object.

The tests are passing

image

File: libraries/bot-integration-core/src/test/java/com/microsoft/bot/integration/CloudAdapterTests.java [hunk 2]
```diff
@@ -77,36 +81,6 @@ public void basicMessageActivity() {
         Mockito.verify(botMock, Mockito.times(1)).onTurn(Mockito.any(TurnContext.class));
     }
 
-    @Test
```
Comment (matiasroldan6) [inline libraries/bot-integration-core/src/test/java/com/microsoft/bot/integration/CloudAdapterTests.java:80]: Just to know. Why was this removed?

**Review by ldardick — APPROVED** [review]

**Review by matiasroldan6 — APPROVED** [review]

File: libraries/bot-integration-core/src/test/java/com/microsoft/bot/integration/CloudAdapterTests.java (hunk 2 above)
Comment (Batta32) [reply libraries/bot-integration-core/src/test/java/com/microsoft/bot/integration/CloudAdapterTests.java:80]: All is explained [here](https://southies.slack.com/archives/C01HGUQ4UDU/p1626188521220600).

But, summarizing:
The behavior between Java and C# differs here, C# is mocking a `HttpRequest` which is present in [Conversation.cs](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Connector/Conversations.cs) that Java handle it using an external service `RestClient` used by [RestConversation.java](https://github.com/microsoft/botbuilder-java/blob/main/libraries/bot-connector/src/main/java/com/microsoft/bot/connector/rest/RestConversations.java).

As C# is mocking a call that Java doesn't have because it's handled by the external service, the **test is useless** in Java as it's not validating anything from the `CloudAdapter` migration.

I had some doubts if we should or not remove this, because also C# is useless, we will confirm if it's necessary during the JaCoCo but also during the validation of the samples.
